### PR TITLE
Add --test262-es2015 option to run-tests.py

### PR DESCRIPTION
**Pull request by ossy-szeged**
JerryScript-DCO-1.0-Signed-off-by: Csaba Osztrogonác oszi@inf.u-szeged.hu

### CODE REVIEW

**Comment by clover2123** [comment]
AFAIK there is no distinct es2015 release in test262, so we should select one practical commit that includes es2015 related tests but not any es7+ features.
I'm wondering which version of test262 is chosen for es2015.

File: tools/runners/run-test-suite-test262.py [hunk 1]
```diff
@@ -28,37 +29,86 @@ def get_platform_cmd_prefix():
...
+    if return_code:
+        print('Cloning test262 repository failed.')
+        return return_code
+
+    if args.es2015:
+        git_hash = 'fd44cd73dfbce0b515a2474b7cd505d6176a9eb5'
+    else:
+        git_hash = 'es5-tests'
+
+    proc = subprocess.Popen(['git', 'checkout', git_hash], cwd=args.test_dir)
```
Comment (galpeter) [inline tools/runners/run-test-suite-test262.py:65]: Can't we just use `subprocess.call` here?

File: tools/runners/run-test-suite-test262.py [hunk 2]
```diff
@@ -28,37 +29,86 @@ def get_platform_cmd_prefix():
...
+    proc.wait()
+    if proc.returncode:
+        print('Cloning test262 repository failed - invalid git revision.')
+        return proc.returncode
+
+    if args.es2015:
+        shutil.copyfile(os.path.join('tests', 'test262-es6-excludelist.xml'),
+                        os.path.join(args.test_dir, 'excludelist.xml'))
+
+        proc = subprocess.Popen(['git', 'apply', os.path.join('..', 'test262-es6.patch')], cwd=args.test_dir)
```
Comment (galpeter) [inline tools/runners/run-test-suite-test262.py:75]: `subprocess.call` ?

**Comment by ossy-szeged** [comment]
> AFAIK there is no distinct es2015 release in test262, so we should select one practical commit that includes es2015 related tests but not any es7+ features.
> I'm wondering which version of test262 is chosen for es2015.

@clover2123 Yes, you are correct there is no es2015 release in test262. Additionally there is no release since ES5.1, so we have to options. We should pick a good enough revision which covers almost the spec and doesn't contain newer tests. Or we should pick a newer revision with better ES2015 coverage, but we have to skip tests for ES2015+ specs. I think fd44cd7... (which was suggested by you, thanks for it) is good enough to reach our goals. (The next b517ca71... hash adds an ES8 feature)

Additionally my goal was not to modify test262 at all. Unfortunately it wasn't possible, because exclude list handling was buggy at this point, because if you added eg. foo.js to exclude list, you excluded all foo.js in the tree, it wasn't possible to specify which one you want to skip. Apart from this minor bugfix, we can use the runner in the test262 repository. But in the future we will need new runner, because it isn't maintained and was removed from tree in Aug 2016.

File: tools/runners/run-test-suite-test262.py (hunk 1 above)
Comment (ossy-szeged) [reply tools/runners/run-test-suite-test262.py:65]: I wanted to use it, but unfortunately only Popen supports cwd argument.

File: tools/runners/run-test-suite-test262.py (hunk 2 above)
Comment (ossy-szeged) [reply tools/runners/run-test-suite-test262.py:75]: ditto

File: tools/runners/run-test-suite-test262.py (hunk 1 above)
Comment (galpeter) [reply tools/runners/run-test-suite-test262.py:65]: `call` supports the same arguments as the Popen.

Comment (ossy-szeged) [reply tools/runners/run-test-suite-test262.py]: Ah, good to know. I should have read the manual thoroughly.

**Pull request merged**